vizro-core/examples/kpi/utils/_charts.py

```diff
@@ -0,0 +1,198 @@
+
+import dash_bootstrap_components as dbc
+import pandas as pd
+import vizro.models as vm
+import vizro.plotly.express as px
+from dash import html
+from vizro.models.types import capture
+
+
+# CUSTOM COMPONENTS -------------------------------------------------------------
+# Note: This is a static KPI Card only (it will not be reactive to controls). A new dynamic KPI Card component
+# is currently in development.
+class KPI(vm.VizroBaseModel):
+    """Static custom `KPI` Card."""
+
+    type: Literal["kpi"] = "kpi"
+    title: str
+    value: str
+    icon: str
+    sign: Literal["delta-pos", "delta-neg"]
+    ref_value: str
+
+    def build(self):
+        return dbc.Card(
+            [
+                html.H4(self.title),
+                html.P(self.value),
+                html.Span(
+                    [
+                        html.Span(self.icon, className="material-symbols-outlined"),
+                        html.Span(self.ref_value),
+                    ],
+                    className=self.sign,
+                ),
+            ],
+            className="kpi-card-ref",
+        )
+
+
+# CUSTOM CHARTS ----------------------------------------------------------------
+@capture("graph")
+def bar(  # noqa: PLR0913
+    x: str,
+    y: str,
+    data_frame: pd.DataFrame,
+    top_n: int = 15,
+    color_discrete_sequence: Optional[List[str]] = None,
+    custom_data: Optional[List[str]] = None,
+):
+    df_agg = data_frame.groupby([y]).aggregate({x: "count"}).sort_values(by=x, ascending=False).reset_index()
+
+    fig = px.bar(
+        data_frame=df_agg.head(top_n),
+        x=x,
+        y=y,
+        orientation="h",
+        text=x,
+        color_discrete_sequence=color_discrete_sequence,
+        custom_data=custom_data,
+    )
+    fig.update_layout(xaxis_title="# of Complaints", yaxis=dict(title="", autorange="reversed"))  # noqa: C408
+    return fig
+
+
+@capture("graph")
+def line(x: str, y: str, data_frame: pd.DataFrame, color_discrete_sequence: Optional[List[str]] = None):
+    df_agg = data_frame.groupby([x]).aggregate({y: "count"}).reset_index()
+    fig = px.area(
+        data_frame=df_agg,
+        x=x,
+        y=y,
+        color_discrete_sequence=color_discrete_sequence,
+        title="Complaints over time",
+    )
+    fig.update_layout(xaxis_title="Date Received", yaxis_title="# of Complaints", title_pad_t=4)
+    return fig
+
+
+@capture("graph")
+def pie(  # noqa: PLR0913
+    names: str,
+    values: str,
+    data_frame: pd.DataFrame = None,
+    color_discrete_sequence: Optional[List[str]] = None,
+    title: Optional[str] = None,
+    custom_order: Optional[List[str]] = None,
+):
+    df_agg = data_frame.groupby([names]).aggregate({values: "count"}).reset_index()
+
+    # Apply custom order
+    order_mapping = {category: index for index, category in enumerate(custom_order)}
+    df_agg["order"] = df_agg[names].map(order_mapping)
+    df_sorted = df_agg.sort_values("order")
```

Comment:
You should be able to do this without creating the column `df_agg["order"]` itself (assuming it's not needed for anything else, which I don't think it is. Something like this:
```
df_sorted = df_agg.sort_values(by=names, key: lambda names: names.map(order_mapping)
```